src/modules/shared/components/Navigation/NavigationList/NavigationList.tsx

```diff
@@ -1,49 +1,90 @@
+import { NavigationDropdown } from '../NavigationDropdown';
 
 import { NavigationListProps } from './NavigationList.types';
 
 const NavigationList: FC<NavigationListProps> = ({ items }) => {
+	const [openDropdown, setOpenDropdown] = useState('');
+
+	const renderDropdown = (
+		id: string,
+		trigger: ReactNode,
+		dropdownItems: NavigationItem[]
+	): ReactNode => {
+		return (
+			<NavigationDropdown
+				id={id}
+				isOpen={openDropdown === id}
+				items={dropdownItems}
+				trigger={trigger}
+				lockScroll
+				onOpen={(id) => setOpenDropdown(id)}
+				onClose={() => setOpenDropdown('')}
+				flyoutClassName={styles['c-navigation__list-flyout']}
+			/>
+		);
+	};
+
+	const renderTrigger = (item: NavigationItem, iconName: IconLightNames) => {
+		return (
+			<div className={styles['c-navigation__link--wrapper']}>
+				{item.node}
+				<Icon className="u-text-left u-ml-4" name={iconName} />
+			</div>
+		);
+	};
+
 	return (
 		<>
-			{items.map((navItems, itemIndex) => {
-				return (
-					<ul key={`nav-list-${itemIndex}`} className={styles['c-navigation__list']}>
-						{navItems.map((item, i) => {
-							const linkCls = clsx(
-								styles['c-navigation__link'],
-								styles[`c-navigation__link--variant-${i + 1}`],
-								{
-									[styles['c-navigation__link--active']]: item.isActive,
-								}
-							);
-
-							return (
-								<li key={item.label} className={styles['c-navigation__item']}>
-									<Link href={item.href}>
-										<a className={linkCls}>
-											{item.label}
-											<span
-												className={
-													styles['c-navigation__border-decoration']
-												}
-											/>
-											<span
-												className={
-													styles['c-navigation__border-decoration']
-												}
-											/>
-										</a>
-									</Link>
-								</li>
-							);
-						})}
-					</ul>
-				);
-			})}
+			<Overlay
+				visible={!!openDropdown}
+				className={clsx(
+					styles['c-navigation__dropdown-overlay'],
+					styles['c-navigation__list-overlay']
+				)}
+			/>
+			<ul className={styles['c-navigation__list']}>
+				{items.map((item, index) => {
+					return (
+						<>
+							{item.hasDivider && (
+								<div className={styles['c-navigation__divider--vertical']} />
+							)}
+							<li
+								key={`navigation-item-${index}`}
+								className={clsx(
+									styles['c-navigation__item'],
+									styles[`c-navigation__link--variant-${index + 1}`],
+									item.active && [styles['c-navigation__item--active']]
+								)}
+							>
+								{item.children?.length
+									? renderDropdown(
+											item.id,
+											renderTrigger(
+												item,
+												openDropdown === item.id ? 'angle-down' : 'angle-up'
+											),
+											item.children
+									  )
+									: item.node}
+								{item.active && (
+									<span className={styles['c-navigation__border-decoration']} />
+								)}
+								{item.active && (
```

Comment:
Is het de bedoeling dat er hier 2 staan, zo ja dan kan je die `{item.active && ...}` combineren.

Reply:
Fixed!